tests/testmplex.nim

```diff
@@ -135,18 +135,20 @@ suite "Mplex":
       let
         conn = newBufferStream(
           proc (data: seq[byte]) {.gcsafe, async.} =
-            discard
+            discard,
+          timeout = 5.minutes
         )
         chann = LPChannel.init(1, conn, true)
 
       await chann.pushTo(("Hello!").toBytes)
-      let closeFut = chann.closeRemote()
 
       var data = newSeq[byte](6)
-      await chann.readExactly(addr data[0], 6) # this should work, since there is data in the buffer
+      await chann.readExactly(addr data[0], 3) # this should work, since there is data in the buffer
```

Comment:
Can you add a test where you write to a stream after the other end has half-closed it? This will ensure that we won't get an unnoticed regression for the usage pattern in the Eth2 syncing protocol.